### PR TITLE
refactor(lint): make linter super happy

### PR DESCRIPTION
**Pull request by ferferga**
Remaining warnings are fixed in #396 and #382

### CODE REVIEW

File: pages/item/_itemId/index.vue [hunk 1]
```diff
@@ -144,6 +145,7 @@
                       single-line
                       hide-details
                     >
+                      <!-- eslint-disable-next-line vue/no-template-shadow -->
```
Comment (Camc314) [inline pages/item/_itemId/index.vue:148]: There must be a better way to fix this

File: pages/library/_viewId.vue [hunk 2]
```diff
@@ -48,10 +48,10 @@ export default Vue.extend({
       sortBy: 'SortName',
       hasFilters: false,
       isDefaultView: true, // Movie view, not Collection. Music view, not Genres...
-      statusFilter: [],
-      genresFilter: [],
-      yearsFilter: [],
-      ratingsFilter: [],
+      statusFilter: [] as unknown,
```
Comment (Camc314) [inline pages/library/_viewId.vue:51]: Same here. Isn't it an array of strings?

Comment (ferferga) [reply pages/library/_viewId.vue:51]: @Camc314 I was not sure but all I checked was indeed a string.

If you're sure it's always a string then I can change it :).

File: pages/item/_itemId/index.vue (hunk 1 above)
Comment (ferferga) [reply pages/item/_itemId/index.vue:148]: @Camc314 I changed the variable's name and it's references and It doesn't worked, everything was undefined. I'm not really sure what ``slot-proposition`` does at all, maybe @mrtimscampi can shed some light on this.

File: pages/library/_viewId.vue (hunk 2 above)
Comment (Camc314) [reply pages/library/_viewId.vue:51]: Yep, just double checked, it is an array of strings

File: pages/item/_itemId/index.vue (hunk 1 above)
Comment (MrTimscampi) [reply pages/item/_itemId/index.vue:148]: You can rename variable while destructuring.

`slot-scope` defines variables that are passed to the scope. So here, we pass `items` to the `v-select`, which then iterates through them. Its template has a slot, into which is passe the item. That's what you're getting in `slot-scope`.

You can do this to rename the variable:
```vue
<template slot="selection" slot-scope="{ item: mediaSourceItem }">
```

**Review by MrTimscampi — APPROVED** [review]

**Pull request merged**